backend/app/controllers/users_controller.rb

```diff
@@ -4,12 +4,39 @@ class UsersController < ApplicationController
   def index
     @users = policy_scope(User).order(:email)
 
-    if users_with_roles_only?
+    users_with_roles_only = params.fetch(:users_with_roles_only, nil)
+
+    if users_with_roles_only == "true"
       @users = @users.with_at_least_one_role
     end
 
+    filter = params.fetch(:filter, nil)
+    puts filter.inspect
+
+    if filter.present?
+      begin
+        filter_email = JSON.parse(filter)
+        filter_email.each do |email|
+          filter_value = email.values.join(" , ")
```

Comment:
not sure what this do. I do not think there should be anything to join here since the value should contain only an email.